tket2/src/serialize/pytket/encoder.rs

```diff
@@ -602,3 +648,43 @@ impl ParameterTracker {
         self.parameters.get(wire)
     }
 }
+
+/// A utility class for finding new unused qubit/bit names.
+#[derive(Debug, Clone, Default)]
+struct RegisterUnitGenerator {
+    /// The next index to use for a new register.
+    next_unit: u16,
+    /// The register name to use.
+    register: String,
+}
+
+impl RegisterUnitGenerator {
+    /// Create a new [`RegisterUnitGenerator`]
+    ///
+    /// Scans the set of existing registers to find the last used index, and
+    /// starts generating new unit names from there.
+    pub fn new<'a>(
+        register: impl ToString,
+        existing: impl IntoIterator<Item = &'a RegisterUnit>,
+    ) -> Self {
+        let register = register.to_string();
+        let mut last_unit: Option<u16> = None;
+        for reg in existing {
+            if reg.0 != register {
+                continue;
+            }
+            last_unit = Some(reg.1[0] as u16);
```

Comment:
does this assume the `existing` are always returned in order (i.e. q[0] is checked before q[1])?

Reply:
🤦Yep